### PR TITLE
RFC: Adds a Padua Transform

### PR DESCRIPTION
**Pull request by MikeAClarke**

### CODE REVIEW

**Comment by dlfivefifty** [comment]
You need to make sure that PaduaTransforms.jl is included in src/FastTransforms.jl, and that the tests are run in runtests.jl

**Comment by MikeAClarke** [comment]
Hi Mikael, thank you for your suggestions.  I've made changes to address all of them except for the question of interpolation on a set of points not touching the boundaries (which I'm currently working on).  The following are the main modifications that were made:

- In the run tests, specifically the norm testing part, the higher tolerance test `@test_approx_eq `was used in lieu of the L2-norm since as you said it is more of a sanity check. Run time tests were added for the pre-planned transforms.

- The `paduapoints ` function was rewritten to avoid constructing separate 2d arrays and improve efficiency.  Also `sinpi()` replaced `cos() ` leading to better accuracy for the coordinates as you suggested.

- Similarly the functions `paduavec `and `paduavalsmat `were rewritten to avoid using various inefficient inbuilt functions like `deleteat!` which improved run times.

- The pre-planned transforms were added via the functions `plan_(i)paduatransform.`  This has been added as an argument `P::(I)PaduaTransformPlan` in a second method for `(i)paduatransform` and each of the functions used within them.  In various run time testing and profiling I've noticed that the pre-planned transforms as currently implemented are not significantly more efficient than the original transforms.  One contributing factor seemed to be the number (and type) of variables added to the plans - in particular the pre-allocated arrays used each function.  So it seems that my implementation of the plans isn't yet at an optimal stage.  Thanks for your consideration.

**Comment by MikaelSlevinsky** [comment]
Super, thanks!

File: src/PaduaTransform.jl [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    tensorvals::Matrix{Float64}
+    padvals::Vector{Float64}
+    dctplan::DCTPLAN
+end
+
+function plan_ipaduatransform(v::AbstractVector)
+    N=length(v)
+    n=Int(cld(-3+sqrt(1+8N),2))
+    @assert N==div((n+1)*(n+2),2)
+    IPaduaTransformPlan(zeros(Float64,n+2,n+1),Array(Float64,n+2,n+1),
```
Comment (dlfivefifty) [inline src/PaduaTransform.jl]: Replace `zeros` with `Array` and use `fill!`.  

Also, use `eltype(v)` instead of `Float64`.

Comment (dlfivefifty) [inline src/PaduaTransform.jl]: Add support for types other than `Float64`

**Pull request merged**